pyabc/visualization/sample.py

```diff
@@ -83,3 +83,89 @@ def plot_sample_numbers(
     fig.tight_layout()
 
     return ax
+
+
+def plot_total_sample_numbers(
+        histories: Union[List, History],
+        labels: Union[List, str] = None,
+        rotation: int = 0,
+        title: str = "Total required samples",
+        yscale: str = 'lin',
+        size: tuple = None,
+        ax: mpl.axes.Axes = None):
+    """
+    Bar plot of total required sample number over all iterations, i.e.
+    a single-colored bar per history, in contrast to `plot_sample_numbers`,
+    which visually distinguishes iterations.
+
+    Parameters
+    ----------
+
+    histories: Union[List, History]
+        The histories to plot from. History ids must be set correctly.
+    labels: Union[List ,str], optional
+        Labels corresponding to the histories. If None are provided,
+        indices are used as labels.
+    rotation: int, optional (default = 0)
+        Rotation to apply to the plot's x tick labels. For longer labels,
+        a tilting of 45 or even 90 can be preferable.
+    title: str, optional (default = "Total required samples")
+        Title for the plot.
+    yscale: str, optional (default = 'lin')
+        The scale on which to plot the counts. Can be one of 'lin', 'log'
+        (basis e) or 'log10'
+    size: tuple of float, optional
+        The size of the plot in inches.
+    ax: matplotlib.axes.Axes, optional
+        The axis object to use.
+
+    Returns
+    -------
+
+    ax: Axis of the generated plot.
+    """
+    # preprocess input
+    histories, labels = to_lists_or_default(histories, labels)
+
+    # create figure
+    if ax is None:
+        fig, ax = plt.subplots()
+    else:
+        fig = ax.get_figure()
+
+    n_run = len(histories)
+
+    # extract sample numbers
+    samples = []
+    for history in histories:
+        # note: the first entry corresponds to the calibration and should
+        # be included here to be fair against methods not requiring
+        # calibration
+        samples.append(np.sum(history.get_all_populations()['samples']))
+    samples = np.array(samples)
+
+    # apply scale
+    ylabel = "Total samples"
+    if yscale == 'log':
+        samples = np.log(samples)
+        ylabel = "log(" + ylabel + ")"
```

Comment:
is it normally used? plotting axes in natural log?

Reply:
don't know, i never did :smile: but doesn't hurt to allow it.